### PR TITLE
Add interface verification with -verify-emitted-module-interface

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Jobs/Job.swift [hunk 1]
```diff
@@ -185,6 +186,9 @@ extension Job : CustomStringConvertible {
 
     case .scanClangDependencies:
       return "Scanning dependencies for Clang module \(moduleName)"
+
+    case .verifyModuleInterface:
+      return "Verifying emitted module interface for module: \(moduleName)"
```
Comment (DougGregor) [inline Sources/SwiftDriver/Jobs/Job.swift:191]: Okay, this is minor, but this message has a colon before the module name, unlike the preceding job kind. Can we make them consistent?

**Review by DougGregor — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**